libwebauthn/src/pin.rs

```diff
@@ -46,9 +46,22 @@ impl Default for PinUvAuthToken {
     }
 }
 
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum PinRequestReason {
+    Direct,
+    FallbackFromUV,
+    // Passkey
+}
+
 #[async_trait]
 pub trait PinProvider: Send + Sync {
-    async fn provide_pin(&self, attempts_left: Option<u32>) -> Option<String>;
+    async fn provide_pin(
+        &self,
+        attempts_left: Option<u32>,
+        reason: PinRequestReason,
+    ) -> Option<String>;
+    /// This is for displaying attempts_left only. No direct feedback from the app expected/required.
+    async fn notify_uv_failed(&self, attempts_left: Option<u32>);
```

Comment:
Naming nit: Thinking about this further, if we only emit this upon retrying, this should trigger a UI message to prompt the user to make a new UV attempt. 

Maybe `prompt_uv_retry`?

Cc @iinuwa